pkg/clusteroperator/clusteroperators.go

```diff
@@ -0,0 +1,273 @@
+package clusteroperator
+
+import (
+	"context"
+	"time"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	v1 "github.com/openshift/api/config/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"fmt"
+)
+
+// CoListBuilder provides struct for ClusterOperator List.
+type CoListBuilder struct {
```

Comment:
A couple of comments:
1. Do you see value in creating a dedicated struct for this? I think it may be easier just to change the methods in this file to functions and just use the Discover function in each of the other functions.
2. In my experience it is a lot easier to work with []ClusterOperatorBuilder rather than []ClusterOperator. That way you can add methods to ClusterOperatorBuilder struct and then you can just call them as you iterate over []ClusterOperatorBuilder . For example, your code changes to:
```
func WaitForAllAvailable() {
   for _, clusterOperatorBuilder := range clusterOperatorBuilderList {
     _ = wait.PollImmediate(fiveScds, availableDuration, func() (done bool, err error) {
        return clusterOperatorBuilder.IsAvailable(), nil
     }
}
```
WDYT? Take a look at the [nodes](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/nodes/nodes.go) package, it is designed similarly

Reply:
I did some refactoring + some new methods added